### PR TITLE
feat: blog migrate from terasology.org

### PR DESCRIPTION
**Pull request by ryuk156**
This PR migrates the blog from terasology.org I have migrated the 10 blogs from the live site will be migrating other blogs in a short time, Since I'm working on another PR.
This PR also fixes the issue of page routing on the blog and module page.

Fixing in next PR:
- [X] visuals of next and previous button on the blog page
- [ ] migrating remaining blogs
- [X] changing card layout to centralize all card over the blog page
- [X] fixing home page recent blogs card layout

### CODE REVIEW

**Review by jdrueckert — CHANGES_REQUESTED** [review]
I won't re-read every blog post, but trust in that you didn't mess with the contents in any way :sweat_smile: 
The only issue that caught my eye was that the TeraSaturday tag does not show up in the list of filters.

From our conversation on Discord:

> I believe it would be nice to have a link to the full list of blogs below the recent ones on the Home page, otherwise it might seem as if there are only 4 in total.
Also "Recent News" might be a better fit for the heading than "Blog" if it's only the recent ones.
for the height differences on the home page we definitely have to think of something how to work against that... maybe some notion of min-height?

All of that would also be fine if it's done in a follow-up PR, so I'll leave it up to you which of the aspects you'll address in this PR and which in a following one.

**Comment by ryuk156** [comment]
> I won't re-read every blog post, but trust in that you didn't mess with the contents in any way 
> The only issue that caught my eye was that the TeraSaturday tag does not show up in the list of filters.
> 
> From our conversation on Discord:
> 
> > I believe it would be nice to have a link to the full list of blogs below the recent ones on the Home page, otherwise it might seem as if there are only 4 in total.
> > Also "Recent News" might be a better fit for the heading than "Blog" if it's only the recent ones.
> > for the height differences on the home page we definitely have to think of something how to work against that... maybe some notion of min-height?
> 
> All of that would also be fine if it's done in a follow-up PR, so I'll leave it up to you which of the aspects you'll address in this PR and which in a following one.


Added the Terasaturday tag and changed the blog heading to recent news from will be doing all the changes in the new PR please take a look

File: src/components/Home/Index.css [hunk 1]
```diff
@@ -39,6 +39,7 @@ iframe {
 }
 
 .row_shadow {
+
```
Comment (majordwarf) [inline src/components/Home/Index.css]: Empty Line

File: src/components/Home/Index.jsx [hunk 2]
```diff
@@ -138,6 +138,24 @@ const Index = () => {
           ))}
         </Row>
       </Col>
+
```
Comment (majordwarf) [inline src/components/Home/Index.jsx:139]: Empty Line

**Review by majordwarf — CHANGES_REQUESTED** [review]
Please do the following changes, had a look at the PR, looks good to me. 👍

File: src/components/Home/Index.css (hunk 1 above)
Comment (ryuk156) [reply src/components/Home/Index.css]: Will format it in last commit😅

File: src/components/BlogSearchForm/BlogSearchForm.jsx [hunk 3]
```diff
@@ -30,6 +30,7 @@ const SearchForm = ({ query, filter }) => (
             <option value="">All</option>
             <option value="gsoc">Google Summer Of Code</option>
             <option value="project">Project</option>
+            <option value="gsoc">TeraSaturday</option>
```
Comment (jdrueckert) [inline src/components/BlogSearchForm/BlogSearchForm.jsx]:
```suggestion
            <option value="terasaturday">TeraSaturday</option>
```

**Review by jdrueckert — CHANGES_REQUESTED** [review]
Found a copy-paste error :wink: 

Other than that I noticed a difference in the distance between text and title in the following screenshot (especially comparing the right most cards in the two rows), but I guess that comes from the character limit resulting in fewer required lines in the upper one, correct?
image

**Comment by ryuk156** [comment]
Yes, your are right I have limited  character to 125 I think i don't know the exact length 😅

File: src/components/BlogSearchForm/BlogSearchForm.jsx (hunk 3 above)
Comment (ryuk156) [reply src/components/BlogSearchForm/BlogSearchForm.jsx]: Will fix this 😅

**Review by jdrueckert — DISMISSED** [review]
Looks good to me 👍

File: src/components/PostListing/PostListing.css [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.word-break {
+  hyphens: auto;
```
Comment (majordwarf) [inline src/components/PostListing/PostListing.css:5]: CSS vendor prefixes can help us provide standardized behavior cross browsers.

```suggestion
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
```

**Review by majordwarf — CHANGES_REQUESTED** [review]
Otherwise looks good to me.

File: src/components/PostListing/PostListing.css (hunk 4 above)
Comment (ryuk156) [reply src/components/PostListing/PostListing.css:5]: I was confused to add this or not like  I checked on Mozilla and chrome it works fine

**Review by majordwarf — APPROVED** [review]
LGTM 👍

**Pull request merged**